src/operations/blas3/gemm_local.hpp

```diff
@@ -326,6 +326,32 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
   }
 
  private:
+  /* @brief If beta is not zero then this function will load in values from C,
+  multiply them by the beta value and store them in the results register. If
+  beta is zero then this function does nothing. */
+  template <bool check_m_limit, bool check_n_limit, typename InputPointerType>
+  static SYCL_BLAS_INLINE void preload_result(
+      element_t (&reg_res)[item_rows][item_cols], InputPointerType C,
+      const index_t &ldc, const index_t &mc, const index_t &nc,
+      const element_t &beta, bool out_of_range) {
+    if (out_of_range) {
+      return;
+    }
+    if (!is_beta_zero) {
```

Comment:
You could combine this condition with the previous one, and have an early return, allowing you to reduce indentation